tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp

```diff
@@ -116,6 +116,32 @@ bool RemoveBufferBlockVisitor::updateStorageClass(
     }
   }
 
+  // For pointer-to-struct cases, we may need to update the storage class for
```

Comment:
I know DXC code base is not clean now .. as I mentioned in https://github.com/microsoft/DirectXShaderCompiler/issues/4173, we need refactoring .. It will take us longer and longer time for the maintenance.

Could you please split your code into a separate one?
e.g.,
```
bool RemoveBufferBlockVisitor::updateStorageClass(
    const SpirvType *type, const SpirvType **newType,
    spv::StorageClass *newStorageClass) {
...
if (const auto *innerStructType = dyn_cast<StructType>(innerType)) {
  return updateStorageClassForStruct(...);
}
return false;
}

bool RemoveBufferBlockVisitor::updateStorageClassForStruct(...) {
// Do the same here.
}
```

Reply:
Ended up reusing updateStorageClass for recursing through nested types.